server/src/main/java/io/littlehorse/server/auth/MTLSServerAuthorizer.java

```diff
@@ -0,0 +1,29 @@
+package io.littlehorse.server.auth;
+
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.littlehorse.common.util.CertificateUtil;
+import io.littlehorse.server.listener.MTLSConfig;
+
+public class MTLSServerAuthorizer implements ServerAuthorizer {
+
+    private final String commonName;
+
+    public MTLSServerAuthorizer(MTLSConfig mtlsConfig) {
+        try {
+            this.commonName = CertificateUtil.getCommonNameFromCertificate(mtlsConfig.getCaCertificate());
+        } catch (Exception ex) {
+            throw new IllegalArgumentException("Certificate is not valid", ex);
+        }
+    }
+
+    @Override
+    public <ReqT, RespT> Listener<ReqT> interceptCall(
+            ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+        headers.put(CLIENT_ID, this.commonName);
```

Comment:
If the `commonName` is `null`, should we put `anonymous`? Or is that handled later in the stream @eduwercamacaro ?